app/scripts/lib/channels/inter-tab.js

```diff
@@ -32,14 +39,43 @@ define([
       }
     },
 
+    /**
+     * Register a listener
+     *
+     * @method on
+     * @param {string} name
+     * @param {function} callback
+     *
+     * @return {string} key - key used to unregister a listener
+     */
     on: function (name, callback) {
-      this._crosstab.util.events.on(name, callback);
+      // crosstab registers callbacks by storing functions by a list
+      // of keys. If no key is given, a key is generated by converting the
+      // function to a string. This means a given function can only be
+      // registered once unless a key is also passed in during registration.
+      // Use the current date plus a random number in case two functions are
+      // reigstered in the same millisecond.
+      var key = Date.now() + Math.random();
+      this._crosstab.util.events.on(name, callback, key);
+      return key;
     },
 
-    off: function (name, callback) {
-      this._crosstab.util.events.off(name, callback);
+    /**
+     * Unregister a listener
+     *
+     * @method off
+     * @param {string} name
+     * @param {string} key
+     */
+    off: function (name, key) {
```

Comment:
This was changed from `callback`=>`key` since I realized that is what crosstab wants. I wonder if we should do the conversion to take a `callback` here instead of in the inter-tab-channel-mixin so the inter-tab-channel acts like all the other pub-sub mechanisms we use.